hal/inc/i2c_hal.h

```diff
@@ -44,19 +46,36 @@ typedef enum HAL_I2C_Interface {
     HAL_I2C_INTERFACE3 = 2
 } HAL_I2C_Interface;
 
+/*! I2c Config Structure Version */
+typedef enum HAL_I2C_Config_Version {
+    HAL_I2C_CONFIG_VERSION_1 = 0,
+    HAL_I2C_CONFIG_VERSION_LATEST = HAL_I2C_CONFIG_VERSION_1,
+} HAL_I2C_Config_Version;
+
+typedef struct HAL_I2C_Config {
+    uint16_t size;
+    uint16_t version;
+    uint8_t* rx_buffer;
+    uint32_t rx_buffer_size;
```

Comment:
Prefer using `size_t`. Same to `tx_buffer_size`.

Reply:
There is no real difference and it is fine to affix the 'size' variables to `uint32_t` or any other type if needed.